src/compiler/util/parseJavaScriptParams.js

```diff
@@ -0,0 +1,13 @@
+"use strict";
+
+var ok = require("assert").ok;
+
+function parseJavaScriptParams(params, builder) {
+    ok(typeof params === "string", '"params" should be a string');
+    ok(builder, '"builder" is required');
+
+    var parsed = builder.parseExpression("function(" + params + "){}");
```

Comment:
Would it be better to parse this as an arrow function? Then the line numbers would already be right if we use them eventually.

Reply:
Because `parseExpression` converts to Marko AST and we don't have an arrow function node, it returns a generic `Expression` node (which doesn't have the params attached).  So `function` it is.

Reply:
👍

Reply:
So I was concerned with some of the changes I made to `parseJavaScript.js`.  By using `esprima` directly here I don't need to make those changes and can also use an arrow function.